<launch>
    <node pkg="module-pose-estimation-conventional" type="pose_extraction_node" name="pose_extraction_node" output="screen">
        <param name="itopic_image" value="/d455/color/image_raw" />
        <param name="itopic_depth_image" value="/d455/depth/image_rect_raw" />
        <param name="itopic_depth_camera_info" value="/d455/depth/camera_info" />
        <param name="world_tf_frame_id" value="t265_odom_frame" /> <!-- TODO: FIX -->
        <param name="itopic_detection2D" value="/perception/module_bbox_dummy" /><!-- TODO: Use true bbox from YOLO: "/perception/object_detection/detections" -->
        
        <param name="otopic_PoseWithCovarianceStamped" value="/perception/interactionPointPoseWithCovariance" />

        <param name="img_buffer_size" value="60" />
        <param name="debug" value="1" />
	</node>

    <node pkg="module-pose-estimation-conventional" type="bbox_publisher_node" name="bbox_publisher" output="screen">
        <param name="depth_mean_offset_value" value="-2" />
        <param name="mask_dilation_kernel_size" value="3" />
        <param name="depth_dilation_kernel_size" value="3" />
        
        <param name="itopic_image" value="/d455/color/image_raw" />
        <param name="itopic_depth_image" value="/d455/depth/image_rect_raw" />
        <param name="world_tf_frame_id" value="t265_odom_frame" />
        
        <param name="otopic_detection2D" value="/perception/module_bbox_dummy" />
        
        <param name="debug" value="1" />
    </node>

    <node pkg="rosbag" type="play" name="player" args="-l --clock $(find module-pose-estimation-conventional)/bags/testflight_2021-01-13-14-36-25.bag"/>

</launch>
